src/meta/meta_turtle.c

```diff
@@ -81,12 +93,81 @@ __metadata_load_hot_backup(WT_SESSION_IMPL *session)
         WT_ERR(__wt_getline(session, fs, value));
         if (value->size == 0)
             WT_ERR_PANIC(session, EINVAL, "%s: zero-length value", WT_METADATA_BACKUP);
+        /*
+         * When performing partial backup restore, generate a list of tables that is not part of the
+         * target uri list so that we can drop all entries later. To do this, parse through all the
+         * table metadata entries and check if the metadata entry exists in the target uri hash
+         * table. If the metadata entry doesn't exist in the hash table, append the table name to
+         * the partial backup list.
+         */
+        metadata_key = (char *)key->data;
+        if (F_ISSET(conn, WT_CONN_BACKUP_PARTIAL_RESTORE) &&
+          WT_PREFIX_MATCH(metadata_key, "table:"))
+            /*
+             * The target uri will be the deciding factor if a specific metadata table entry needs
+             * to be dropped. If the metadata table entry does not in the target uri hash table,
+             * append the metadata key to the backup remove list.
+             */
+            if (__wt_backup_target_uri_search(session, metadata_key) == false) {
```

Comment:
This edge case is interesting. So I took a look at the metadata entries for WiredTiger HS table, and found that it only has a `file:` entry and no table metadata entry. In our case, we only parse through table metadata entries, so I think we are fine? @sueloverso let me know your thoughts.

Reply:
That would be fine. Maybe add a comment and assert that the `table:` is not in the WiredTiger namespace. That way if the history store or other WT table ever changes it will trigger and for now it will be fine.